doc/nixpkgs-users-guide.rst

```diff
@@ -166,8 +166,89 @@ for hackage. You can use ``overrideSrc`` to override the source, for example:
 How to create a development environment
 ---------------------------------------
 
+A simple approach for beginners
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The basic workflow
+^^^^^^^^^^^^^^^^^^
+
+1. To bootstrap a brand new project, run this command to create a cabal
+file.
+If you already have a cabal file, you can skip this step.
+
+::
+
+   nix-shell --packages ghc --run 'cabal init'
+
+2. Create ``default.nix`` with the following contents:
+
+::
+
+   let
+     pkgs = import <nixpkgs> { };
+   in
+     pkgs.haskellPackages.developPackage {
+       root = ./.;
+     }
+
+*Important:* Your directory must have the same name as your package,
+or else you have to set the ``name`` field in ``default.nix``.
+
+3. Build your package in one of two ways:
+
+   - Run ``nix-build``.
+   - Run ``nix-shell`` to enter an environment with all of your
+     dependencies available, and then use the usual cabal commands.
+
+Adding or overriding dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you need a package that isn't in Hackage, or you want to use a
+different version, add it to ``default.nix``.
+In the example below, I've added the path to another package named
+``grid``.
+
+::
+
+   let
+     pkgs = import <nixpkgs> { };
+   in
+     pkgs.haskellPackages.developPackage {
+       root = ./.;
+       source-overrides = {
+         grid = ../grid;
```

Comment:
"What is `grid` there", maybe something like `additional-package = ../additional-package`?

Reply:
Change made as suggested/